nordic-nrf51822-armcc/CMake/toolchain.cmake

```diff
@@ -0,0 +1,113 @@
+
+if(YOTTA_CFG_NORDIC_SOFTDEVICE STREQUAL "S110")
+    add_definitions("-DTARGET_MCU_NRF51_${NRF51822_RAM_SIZE} -DTARGET_MCU_NRF51_${NRF51822_RAM_SIZE}_S110")
+    set(MBED_LEGACY_TARGET_DEFINITIONS   ${MBED_LEGACY_TARGET_DEFINITIONS} "MCU_NRF51" "MCU_NRF51_${NRF51822_RAM_SIZE}" "MCU_NORDIC_${NRF51822_RAM_SIZE}_S110")
+    set(NRF51822_LINKER_FLAGS_FILE_PATH  "${CMAKE_CURRENT_LIST_DIR}/../ld/nRF51822_${NRF51822_RAM_SIZE}_S110.sct")
+    set(NRF51822_SOFTDEVICE_FILE_PATH    "${CMAKE_CURRENT_LIST_DIR}/../softdevice/s110_nrf51822_8.0.0_softdevice.hex")
+elseif(YOTTA_CFG_NORDIC_SOFTDEVICE STREQUAL "S130")
+    set(NRF51822_LINKER_FLAGS_FILE_PATH  "${CMAKE_CURRENT_LIST_DIR}/../ld/nRF51822_${NRF51822_RAM_SIZE}_S130.sct")
+    set(NRF51822_SOFTDEVICE_FILE_PATH    "${CMAKE_CURRENT_LIST_DIR}/../softdevice/s130_nrf51_1.0.0_softdevice.hex")
+else()
+    message(FATAL_ERROR "SoftDevice version '${YOTTA_CFG_NORDIC_SOFTDEVICE}' is not recognized. Please check your yotta config file.")
+endif()
+
+# append non-generic flags, and set NRF51822-specific link script
+set(_CPU_COMPILATION_OPTIONS "--CPU=Cortex-M0 -D__thumb2__")
+
+set(CMAKE_C_FLAGS_INIT          "${CMAKE_C_FLAGS_INIT} ${_CPU_COMPILATION_OPTIONS}")
+set(CMAKE_ASM_FLAGS_INIT        "${CMAKE_ASM_FLAGS_INIT} ${_CPU_COMPILATION_OPTIONS}")
+set(CMAKE_CXX_FLAGS_INIT        "${CMAKE_CXX_FLAGS_INIT} ${_CPU_COMPILATION_OPTIONS}")
+set(CMAKE_EXE_LINKER_FLAGS_INIT "${CMAKE_EXE_LINKER_FLAGS_INIT} --info=totals --list=.link_totals.txt --scatter ${NRF51822_LINKER_FLAGS_FILE_PATH}")
+
+# used by the apply_target_rules function below:
+set(NRF51822_SOFTDEVICE_HEX_FILE "${NRF51822_SOFTDEVICE_FILE_PATH}")
+set(NRF51822_BOOTLOADER_HEX_FILE "${CMAKE_CURRENT_LIST_DIR}/../bootloader/s130_nrf51_1.0.0_bootloader.hex")
+set(NRF51822_MERGE_HEX_SCRIPT    "${CMAKE_CURRENT_LIST_DIR}/../scripts/merge_hex.py")
+set(NRF51822_MEMORY_INFO_SCRIPT  "${CMAKE_CURRENT_LIST_DIR}/../scripts/memory_info.py")
+
+# define a function for yotta to apply target-specific rules to build products,
+# in our case we need to convert the built elf file to .hex, and add the
+# pre-built softdevice:
+function(yotta_apply_target_rules target_type target_name)
+    if(${target_type} STREQUAL "EXECUTABLE")
+        add_custom_command(OUTPUT ${target_name}.hex ${target_name}-combined.hex
+            DEPENDS ${target_name}
+            # fromelf to hex
+            COMMAND fromelf --i32combined --output=${target_name}.hex ${target_name}
+            # and append the softdevice hex file
+            COMMAND python ${NRF51822_MERGE_HEX_SCRIPT} ${NRF51822_SOFTDEVICE_HEX_FILE} ${target_name}.hex ${target_name}-combined.hex
+            COMMENT "hexifying and adding softdevice to ${target_name}"
+            VERBATIM
+        )
+        if(YOTTA_CFG_IMAGE_FOTA)
+            add_custom_command(OUTPUT ${target_name}-combined-fota.hex
```

Comment:
I am not entirely sure what is the best way in CMAKE to implement this additional step to produce the fota image. The problem is that this command must be done strictly AFTER the `${target_name}.hex` file is created. So I this is why I have done it this way and modified the command in line [87](https://github.com/ARMmbed/target-nrf51822/pull/32/files#diff-d44ed46023206f5d88041cfbfdc950f6R87) that no longer is `POST_BUILD`.